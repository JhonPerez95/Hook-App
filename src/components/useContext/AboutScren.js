import React from 'react';

const AboutScren = () => {
  return (
    <div>
      <h1>AboutScren</h1>
      <hr />
    </div>
  );
};

export default AboutScren;
