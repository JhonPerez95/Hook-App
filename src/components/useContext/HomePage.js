import React, { useContext } from 'react';

import { UserContext } from './UserContext';

const HomePage = () => {
  const { user } = useContext(UserContext);

  console.log(user);

  return (
    <div>
      <h1>HomePage</h1>
      <hr />
      <pre className="container">{JSON.stringify(user, null, 3)}</pre>
    </div>
  );
};

export default HomePage;
