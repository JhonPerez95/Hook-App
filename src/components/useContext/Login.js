import React, { useContext } from 'react';

import { UserContext } from './UserContext';

const Login = () => {
  const { setUser } = useContext(UserContext);

  const newUser = {
    id: 123131321,
    user: 'jhonPerez95',
    email: 'jhonperez@gmail.com',
  };
  return (
    <div>
      <h1>Login</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(newUser)}>
        Add User
      </button>
    </div>
  );
};

export default Login;
