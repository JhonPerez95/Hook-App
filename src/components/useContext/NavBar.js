import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div className="navbar-nav">
          <Link className="nav-item nav-link active" to="/">
            Home
          </Link>
          <Link className="nav-item nav-link" to="/login">
            Login
          </Link>
          <Link className="nav-item nav-link" to="/about">
            About
          </Link>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
