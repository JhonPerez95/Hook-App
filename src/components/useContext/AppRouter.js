import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';

import HomePage from './HomePage';
import AboutScren from './AboutScren';
import Login from './Login';
import NavBar from './NavBar';

const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/about" component={AboutScren} />
          <Route exact path="/login" component={Login} />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
