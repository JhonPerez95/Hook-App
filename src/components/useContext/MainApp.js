import React, { useState } from 'react';

import AppRouter from './AppRouter';
import { UserContext } from './UserContext';

const MainApp = () => {
  // const user2 = {
  //   id: 123,
  //   name: 'jhon',
  //   email: 'jhon@gmail.com',
  // };

  const [user, setUser] = useState({});
  return (
    <UserContext.Provider value={{ user, setUser }}>
      <AppRouter />
    </UserContext.Provider>
  );
};

export default MainApp;
