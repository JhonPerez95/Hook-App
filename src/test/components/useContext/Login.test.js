import React from 'react';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/useContext/UserContext';
import Login from '../../../components/useContext/Login';

describe('Test the component <Login/>', () => {
  const setUser = jest.fn();
  const newUser = {
    id: 123131321,
    user: 'jhonPerez95',
    email: 'jhonperez@gmail.com',
  };

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <Login />
    </UserContext.Provider>
  );

  test('Debe mostrar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe ejecutar setUser con argumentos esperados', () => {
    wrapper.find('button').simulate('click');
    wrapper.find('button').prop('onClick')();
    expect(setUser).toHaveBeenCalledWith(newUser);
  });
});
