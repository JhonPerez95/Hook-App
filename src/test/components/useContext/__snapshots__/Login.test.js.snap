// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component <Login/> Debe mostrar correctamente 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Add User
  </button>
</div>
`;
